### DISCUSSION TITLE
Error during Deployment

### DISCUSSION THREAD

**Discussion by AkhilsWorkshop** [vercel/next.js#47386] [Help] [answered]
### Summary

I'm new here, so not sure with proper formatting.

My next.js app runs perfectly on local host but during deployment I'm getting the following error. 

I'm not sure how to debug it and also, couldn't link a debugger to it.

Any help or suggestion would be really useful to me. TIA.

### Additional information

```js
> Build error occurred
TypeError: Expected a non-empty string
    at picomatch.makeRe (/vercel/path0/node_modules/next/dist/compiled/micromatch/index.js:15:21418)
    at micromatch.makeRe (/vercel/path0/node_modules/next/dist/compiled/micromatch/index.js:15:2615)
    at /vercel/path0/node_modules/next/dist/build/index.js:1720:48
    at Array.map (<anonymous>)
    at /vercel/path0/node_modules/next/dist/build/index.js:1715:135
    at async Span.traceAsyncFn (/vercel/path0/node_modules/next/dist/trace/trace.js:79:20)
    at async Object.build [as default] (/vercel/path0/node_modules/next/dist/build/index.js:73:29)
Error: Command "npm run build" exited with 1
BUILD_UTILS_SPAWN_1: Command "npm run build" exited with 1
```

**Comment by jaydindg** [comment]
I'm having this issue as well, not sure what the root of the problem is what so ever

**Reply by AkhilsWorkshop** [reply]
Have you figured it out?

**Comment by icyJoseph** [comment, marked as answer]
Are you using remote patterns for next/image? https://nextjs.org/docs/api-reference/next/image#remote-patterns @debug-jay @AkhilsWorkshop 

No custom server right?

**Reply by AkhilsWorkshop** [reply]
Yes, I'm getting images from airtable.

**Reply by icyJoseph** [reply]
Are all env vars correctly set?

**Reply by AkhilsWorkshop** [reply]
You're right. Remote patterns were causing trouble. Changing it to domains fixed my issue.

**Comment by CodeMuscle** [comment]
> ### Summary
> I'm new here, so not sure with proper formatting.
> 
> My next.js app runs perfectly on local host but during deployment I'm getting the following error.
> 
> I'm not sure how to debug it and also, couldn't link a debugger to it.
> 
> Any help or suggestion would be really useful to me. TIA.
> 
> ### Additional information
> ```js
> > Build error occurred
> TypeError: Expected a non-empty string
>     at picomatch.makeRe (/vercel/path0/node_modules/next/dist/compiled/micromatch/index.js:15:21418)
>     at micromatch.makeRe (/vercel/path0/node_modules/next/dist/compiled/micromatch/index.js:15:2615)
>     at /vercel/path0/node_modules/next/dist/build/index.js:1720:48
>     at Array.map (<anonymous>)
>     at /vercel/path0/node_modules/next/dist/build/index.js:1715:135
>     at async Span.traceAsyncFn (/vercel/path0/node_modules/next/dist/trace/trace.js:79:20)
>     at async Object.build [as default] (/vercel/path0/node_modules/next/dist/build/index.js:73:29)
> Error: Command "npm run build" exited with 1
> BUILD_UTILS_SPAWN_1: Command "npm run build" exited with 1
> ```
> 
> ### Example
> _No response_

I'm having this same issue. But I'm not getting images from anywhere outside other than the /public folder.

**Comment by sakilanwar12** [comment]
> Build error occurred
--
10:51:00.911 | TypeError: Expected a non-empty string
10:51:00.911 | at picomatch.makeRe (/vercel/path0/node_modules/next/dist/compiled/micromatch/index.js:15:21418)
10:51:00.911 | at micromatch.makeRe (/vercel/path0/node_modules/next/dist/compiled/micromatch/index.js:15:2615)
10:51:00.912 | at /vercel/path0/node_modules/next/dist/build/index.js:1827:54
10:51:00.912 | at Array.map (<anonymous>)
10:51:00.912 | at /vercel/path0/node_modules/next/dist/build/index.js:1822:155
10:51:00.913 | at async Span.traceAsyncFn (/vercel/path0/node_modules/next/dist/trace/trace.js:103:20)
10:51:00.913 | at async build (/vercel/path0/node_modules/next/dist/build/index.js:145:29)
10:51:00.961 | Error: Command "npm run build" exited with 1

I also face this same isssue.

**Comment by VuongNhat021219** [comment]
`const nextConfig = {
  images: {
    remotePatterns: [
      {
        protocol: "https",
        hostname: "**",
      },
    ],
  },
};`
It works for me, accordingly next.js documentation:

**Reply by anhtuanleee** [reply]
tks u so much, it so helpfull.

**Reply by anhtuanleee** [reply]
cám ơn og ✌️

**Reply by oscarabcorona** [reply]
Thanks

**Reply by absolutegravitas** [reply]
dude that allows all domains... not exactly the point of using remotePatterns
